src/ol_infrastructure/applications/ocw_site/snippets/ttl_setup.vcl

```diff
@@ -2,9 +2,17 @@ if (beresp.status == 404) {
   set beresp.ttl = 5m;
   set beresp.http.Cache-Control = "max-age=300";
   return(deliver);
- }
+}
 
-if (bereq.url.path ~ "^/main\.[0-9a-f]+\.(css|js)") {
+if (bereq.url.path ~ ".*(main|common|www|course_v2|instructor_insights|fields|[0-9a-f]+)\.[0-9a-f]+\.(css|js)") {
   set beresp.ttl = 2629743s;  // one month
   set beresp.http.Cache-Control = "max-age=2629743";
- }
+}
+
+if (bereq.url.path ~ ".*\/static_shared\/(images|fonts)\/[a-zA-Z0-9|-|_]+(\.subset)?\.[0-9a-f]+\.(png|jpg|jpeg|svg|ttf|woff|woff2)") {
+  set beresp.ttl = 2629743s;  // one month
+  set beresp.http.Cache-Control = "max-age=2629743";
+} elsif (bereq.url.path ~ ".*\/static_shared\/(images|fonts)\/.*\.(png|jpg|jpeg|svg|ttf|woff|woff2)") {
+  set beresp.ttl = 604800s;  // one week
+  set beresp.http.Cache-Control = "max-age=604800";
+}
```

Comment:
Why are some assets one month and others are one week?

I'm not just asking why, but I'm also trying to anticipate the next engineer or devops who looks at this and wonders "why is this?" -- how will they find out?

Reply:
>Why are some assets one month and others are one week?
I'm not just asking why, but I'm also trying to anticipate the next engineer or devops who looks at this and wonders "why is this?" -- how will they find out?

The hashed static assets are set to one month, because they will always have fresh content.
The unhashed static assets are set to one week, because they may contain stale content, and the fresh copy will only be fetched after a week is over. There are only about 2 images that fall into this category (that could not be hashed because they were in `.ts` files). So a generalized rule is made which would be applicable to any future unhashed assets as well.

More details can be found here:
https://github.com/mitodl/ocw-hugo-themes/issues/1212#issuecomment-1743506321
https://mitodl.slack.com/archives/GQ89D0Z5M/p1696431334601819